From 84f33f231688bb88eac5fc5a7ca302564d223846 Mon Sep 17 00:00:00 2001
From: Herton Ronaldo Krzesinski <herton@mandriva.com.br>
Date: Mon, 1 Jul 2013 02:18:24 +0200
Subject: [PATCH 04/65] acerhk: add compatibility for kernel 2.6.30

Updated for owner removal from struct proc_dir_entry (2.6.30)

rediff against Debian's acerhk 0.5.35-7 package and change the
	entry->owner = THIS_MODULE
removal [1] into versioned ifdefs, as it is still required for older kernels.

[1]	http://git.kernel.org/linus/99b76233803beab302123d243eea9e41149804f3
	http://bugzilla.kernel.org/show_bug.cgi?id=12454

Signed-off-by: Herton Ronaldo Krzesinski <herton@mandriva.com.br>
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -2702,7 +2702,9 @@ static int acerhk_proc_init(void)
     printk(KERN_INFO"acerhk: could not create /proc/driver/acerhk\n");
   }
   else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
     proc_acer_dir->owner = THIS_MODULE;
+#endif
     /* now create several files, first general info ... */
     entry = create_proc_read_entry("info",
                                    0444, proc_acer_dir, acerhk_proc_info, NULL);
@@ -2711,7 +2713,9 @@ static int acerhk_proc_init(void)
       remove_proc_entry("driver/acerhk", NULL);
       retval = 0;
     } else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
       entry->owner = THIS_MODULE;
+#endif
       /* ... last pressed key ... */
       entry = create_proc_read_entry("key",
                                      0444, proc_acer_dir, acerhk_proc_key, NULL);
@@ -2721,7 +2725,9 @@ static int acerhk_proc_init(void)
         remove_proc_entry("driver/acerhk", NULL);
         retval = 0;
       } else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
         entry->owner = THIS_MODULE;
+#endif
         /* ... and led control file */
         entry = create_proc_entry("led", 0222, proc_acer_dir);
         if (entry == NULL) {
@@ -2733,7 +2739,9 @@ static int acerhk_proc_init(void)
         }
         else {
           entry->write_proc = acerhk_proc_led;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
           entry->owner = THIS_MODULE;
+#endif
           /* ... and wireless led controll file */
           entry = create_proc_entry("wirelessled", 0222, proc_acer_dir);
           if (entry == NULL) {
@@ -2746,7 +2754,9 @@ static int acerhk_proc_init(void)
           }
           else {
             entry->write_proc = acerhk_proc_wirelessled;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
             entry->owner = THIS_MODULE;
+#endif
             /* ... and bluetooth led controll file */
             entry = create_proc_entry("blueled", 0222, proc_acer_dir);
             if (entry == NULL) {
@@ -2759,7 +2769,9 @@ static int acerhk_proc_init(void)
               retval = 0;
             } else {
               entry->write_proc = acerhk_proc_blueled;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
               entry->owner = THIS_MODULE;
+#endif
               retval = 1;
 #ifdef ACERDEBUG
               /* add extra file for debugging purposes */
@@ -2776,7 +2788,9 @@ static int acerhk_proc_init(void)
               }
               else {
                 entry->write_proc = acerhk_proc_debug;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
                 entry->owner = THIS_MODULE;
+#endif
                 retval = 1;
               }
 #endif
